### DISCUSSION TITLE
Stack.Item documented but doesn't appear to be implemented?

### DISCUSSION THREAD

**Discussion by zilch42** [Appsilon/shiny.fluent#86] [Q&A] [answered]
Stack.Item appears in the Stack function documentation >

> A Stack Item abstracts those properties that are or can be specifically applied on flexbox's children, like grow and shrink.
> 
> To use a Stack Item in an application, the Stack component should be imported and Stack.Item should be used inside of a Stack. This is done so that the existence of the  Stack Item is inherently linked to the Stack component.

But doesn't appear to be implemented (I can find no reference to it in the package code). Is there some way to use it or is this not possible at the moment?

I would like to be able to set individual stack item styles as in some of the examples on the [Microsoft Fluent UI page](https://developer.microsoft.com/en-us/fluentui#/controls/web/stack)

**Comment by kamilzyla** [comment, marked as answer]
Thanks for asking! Currently shiny.react cannot handle "nested" components, so shiny.fluent doesn't include them. I created issue #87 to track this feature.

For now it is possible to use the following workaround:
```r
library(shiny.fluent)

CustomComponents <- tags$script(HTML("(function() {
  const CustomComponents = jsmodule['CustomComponents'] = {};
  const { Stack } = jsmodule['@fluentui/react'];
  CustomComponents.StackItem = Stack.Item;
})();"))

Stack.Item <- function(...) shiny.react::reactElement(
  module = "CustomComponents",
  name = "StackItem",
  props = shiny.react::asProps(...),
  deps = shiny.fluent::shinyFluentDependency()
)

shinyApp(
  ui = tagList(
    CustomComponents,
    Stack(
      Stack.Item("One", order = 2),
      Stack.Item("Two", order = 1)
    )
  ),
  server = function(input, output) {}
)
```
